streaming/src/main/scala/org/apache/carbondata/streaming/parser/FieldConverter.scala

```diff
@@ -54,11 +54,12 @@ object FieldConverter {
       value match {
         case s: String => if (!isVarcharType && !isComplexType &&
                               s.length > CarbonCommonConstants.MAX_CHARS_PER_COLUMN_DEFAULT) {
-          throw new IllegalArgumentException(stringLengthExceedErrorMsg +
-            CarbonCommonConstants.MAX_CHARS_PER_COLUMN_DEFAULT + " characters")
-        } else {
-          s
+          if (!CarbonProperties.getInstance.getProperty(CarbonCommonConstants
+            .CARBON_ENABLE_BAD_RECORD_HANDLING_FOR_INSERT).toBoolean) {
+            throw new IllegalArgumentException(CarbonCommonConstants.STRING_LENGTH_EXCEEDED_MESSAGE)
```

Comment:
`STRING_LENGTH_EXCEEDED_MESSAGE` has format specifiers("%s") in it. It is directly passed `IllegalArgumentException` without formatting. Can cause security vulnerability issue. Also suggest to add FMT prefix to `STRING_LENGTH_EXCEEDED_MESSAGE` so that we don't miss such issues.

`public static final String STRING_LENGTH_EXCEEDED_MESSAGE =
      "Record %s of column %s exceeded " + MAX_CHARS_PER_COLUMN_DEFAULT +
          " characters. Please consider long string data type.";`